htdocs/avoloicustomtypes/class/api_avoloicustomtypes.class.php

```diff
@@ -233,4 +233,197 @@ public function tierstype($id) {
+
+		$libelle = $eventType;
+		
+		$sql = "UPDATE `llx_c_actioncomm` SET ";
+		$sql.= "`color`='$color_code', ";
+		$sql.= "`libelle`='$libelle' ";
+		$sql.= "WHERE id = $id";
+
+		$resql = $this->db->query($sql);
+
+		// Vérification de la bonne modification de la ligne
+		$sqlTmp = "SELECT * FROM `llx_c_actioncomm` WHERE id = $id";
+		$resqTmp = $this->db->query($sqlTmp);
+		$objTmp = $this->db->fetch_object($resqTmp);
+
+		if ($objTmp->libelle !== $libelle) {
+			throw new RestException(405, "Problème lors de la modification du type de rendez-vous");
+		}
+
+		if ($objTmp->color !== $color_code) {
+			throw new RestException(405, "Problème lors de la modification du type de rendez-vous");
+		}
+
+		return $id;
+	}
+
+	/**
+	 * Get event types
+	 * 
+	 * @return  string			Event types
+	 *
+	 * @throws 500
+	 * @throws 501
+	 * @throws 400
+	 * @throws 401
+	 * @throws 404
+	 * @throws 200
+	 *
+	 * @url GET /eventtype
+	 */
+	public function geteventtypes() {
+		global $conf, $langs, $user;
+		
+		$sql = "SELECT id, code, libelle, color ";
+		$sql.= "FROM `llx_c_actioncomm` WHERE active = 1 ";
+		$sql.= "EXCEPT (SELECT id, code, libelle, color FROM `llx_c_actioncomm` WHERE code = 'AC_OTH_AUTO' OR code = 'AC_OTH')";
+
+		$resql = $this->db->query($sql);
+
+		return $resql;
+	}
+
+	/**
+	 * Delete an event type
+	 * 
+	 * @param		string		$id						Event ID
+	 * @return  string									Created type ID
+	 *
+	 * @throws 500
+	 * @throws 501
+	 * @throws 400
+	 * @throws 401
+	 * @throws 404
+	 * @throws 200
+	 *
+	 * @url DELETE /deleteeventtype
+	 */
+	public function deleteeventtype($id) {
+		global $conf, $langs, $user;
+
+		$sql = "DELETE FROM `llx_c_actioncomm` WHERE `id` = $id;";
+		$resql = $this->db->query($sql);
+
+		// Vérification de la bonne suppression de la ligne
+		$sqlTmp = "SELECT * FROM `llx_c_actioncomm` WHERE id = $id";
+		$resqTmp = $this->db->query($sqlTmp);
+		$objTmp = $this->db->fetch_object($resqTmp);
+
+		// Si le type de rendez-vous avec id = $id existe, la suppression ne s'est pas faite
+		// On renvoit donc une erreur
+		if ($objTmp) {
+			throw new RestException(405, "Problème lors de la suppression du type de rendez-vous");
+		}
+
+		return "Type de rendez-vous supprimé";
+	}
+
+	private function generaterandomcode() {
+		// Caractères pouvant aparaîtres dans le code généré
+		$characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+
+		$code = '';
+
+		// Un caractère de $charaters est sélectionné au hasard
+		$max = strlen($characters) - 1;
+		for ($i = 0; $i < 11; $i++) {
+				 $code .= $characters[mt_rand(0, $max)];
+		}
+
+
+		// Vérification de l'existence de ce code dans la table
+		$sql = "SELECT * FROM `llx_c_actioncomm` WHERE code = '$code';";
+		$resql = $this->db->query($sql);
+		$objTmp = $this->db->fetch_object($resql);
+
+		// Si le code existe dans la table, on rappel la fonction de génération pour proposer un nouveau code aléatoirfe
+		if ($objTmp) {
+			$code = $this->generaterandomcode();
```

Comment:
Cf mon commentaire sur le camelCase

```suggestion
			$code = $this->generateRandomCode();
```